src/domain/opendatafabric/src/identity/dataset_refs.rs

```diff
@@ -740,3 +741,126 @@ impl std::cmp::PartialOrd for DatasetRefAny {
         Some(self.cmp(other))
     }
 }
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum DatasetRefPattern {
+    Ref(DatasetRef),
+    Pattern(Option<AccountName>, DatasetNamePattern),
+}
+
+impl DatasetRefPattern {
+    pub fn match_pattern(dataset_ref: &str, pattern: &str) -> Result<bool, InvalidPatternError> {
+        Like::<false>::like(dataset_ref, pattern)
+    }
+}
+
+impl std::str::FromStr for DatasetRefPattern {
+    type Err = ParseError<Self>;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.contains('%') {
+            return match s.split_once('/') {
+                Some((account, dn)) => match DatasetNamePattern::try_from(dn) {
+                    Ok(dnp) => match AccountName::try_from(account) {
+                        Ok(an) => Ok(Self::Pattern(Some(an), dnp)),
+                        Err(_) => Err(Self::Err::new(s)),
+                    },
+                    Err(_) => Err(Self::Err::new(s)),
+                },
+                None => match DatasetNamePattern::try_from(s) {
+                    Ok(dnp) => Ok(Self::Pattern(None, dnp)),
+                    Err(_) => Err(Self::Err::new(s)),
+                },
+            };
+        }
+        match DatasetRef::from_str(s) {
+            Ok(dr) => Ok(Self::Ref(dr)),
+            Err(_) => Err(Self::Err::new(s)),
+        }
+    }
+}
+
+super::dataset_identity::impl_parse_error!(DatasetRefPattern);
+
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
+
```

Comment:
Much more advanced coverage is necessary to cover patterns grammar.

Reply:
I also don't see any tests for matching itself.